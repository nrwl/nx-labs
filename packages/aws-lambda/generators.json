{
  "$schema": "http://json-schema.org/schema",
  "name": "aws-lamda",
  "version": "0.0.1",
  "generators": {
    "serverless": {
      "factory": "./src/generators/serverless/serverless",
      "schema": "./src/generators/serverless/schema.json",
      "description": "Creates a new project with AWS Lambda configuration"
    },
    "setup-serverless": {
      "factory": "./src/generators/setup-serverless/setup-serverless",
      "schema": "./src/generators/setup-serverless/schema.json",
      "description": "Adds AWS Lamda Serverless configuration to an existing project"
    },
    "preset": {
      "factory": "./src/generators/preset/preset",
      "schema": "./src/generators/preset/schema.json",
      "description": "Nx AWS Lambda Serverless preset generator",
      "hidden": true
    }
  }
}
