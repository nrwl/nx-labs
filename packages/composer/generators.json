{
  "generators": {
    "init": {
      "factory": "./src/generators/init/init",
      "schema": "./src/generators/init/schema.json",
      "description": "Initializes Composer in the current workspace",
      "hidden": true
    },
    "library": {
      "factory": "./src/generators/library/library",
      "schema": "./src/generators/library/schema.json",
      "description": "library generator"
    }
  }
}
