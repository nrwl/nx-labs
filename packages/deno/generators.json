{
  "$schema": "http://json-schema.org/schema",
  "name": "deno",
  "version": "0.0.1",
  "generators": {
    "application": {
      "factory": "./src/generators/application/generator",
      "schema": "./src/generators/application/schema.json",
      "description": "Create a new Deno application",
      "aliases": ["app"]
    },
    "library": {
      "factory": "./src/generators/library/library",
      "schema": "./src/generators/library/schema.json",
      "description": "Create a new Deno library",
      "aliases": ["lib"]
    },
    "init": {
      "factory": "./src/generators/init/generator",
      "schema": "./src/generators/init/schema.json",
      "description": "Intialize configurations for Deno"
    },
    "preset": {
      "factory": "./src/generators/preset/generator",
      "schema": "./src/generators/preset/schema.json",
      "description": "Create a new standalone Deno workspace with create-nx-workspace"
    },
    "add-import": {
      "aliases": ["import", "add"],
      "factory": "./src/generators/add-import/generator",
      "schema": "./src/generators/add-import/schema.json",
      "description": "Add an existing library to Deno imports"
    }
  }
}
