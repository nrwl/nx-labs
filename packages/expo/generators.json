{
  "name": "Nx Expo",
  "version": "0.1",
  "extends": ["@nrwl/workspace"],
  "schematics": {
    "init": {
      "factory": "./src/generators/init/init#expoInitSchematic",
      "schema": "./src/generators/init/schema.json",
      "description": "Initialize the @nrwl/expo plugin",
      "hidden": true
    },
    "application": {
      "factory": "./src/generators/application/application#expoApplicationSchematic",
      "schema": "./src/generators/application/schema.json",
      "aliases": ["app"],
      "x-type": "application",
      "description": "Create an application"
    },
    "library": {
      "factory": "./src/generators/library/library#expoLibrarySchematic",
      "schema": "./src/generators/library/schema.json",
      "aliases": ["lib"],
      "x-type": "library",
      "description": "Create a library"
    },
    "component": {
      "factory": "./src/generators/component/component#expoComponentSchematic",
      "schema": "./src/generators/component/schema.json",
      "description": "Create a component",
      "aliases": ["c"]
    }
  },
  "generators": {
    "init": {
      "factory": "./src/generators/init/init#expoInitGenerator",
      "schema": "./src/generators/init/schema.json",
      "description": "Initialize the @nrwl/expo plugin",
      "hidden": true
    },
    "application": {
      "factory": "./src/generators/application/application#expoApplicationGenerator",
      "schema": "./src/generators/application/schema.json",
      "aliases": ["app"],
      "x-type": "application",
      "description": "Create an application"
    },
    "library": {
      "factory": "./src/generators/library/library#expoLibraryGenerator",
      "schema": "./src/generators/library/schema.json",
      "aliases": ["lib"],
      "x-type": "library",
      "description": "Create a library"
    },
    "component": {
      "factory": "./src/generators/component/component#expoComponentGenerator",
      "schema": "./src/generators/component/schema.json",
      "description": "Create a component",
      "aliases": ["c"]
    }
  }
}
