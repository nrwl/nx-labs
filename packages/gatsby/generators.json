{
  "name": "nx/gatsby",
  "version": "0.1",
  "extends": ["@nrwl/react"],
  "schematics": {
    "init": {
      "factory": "./src/generators/init/init#gatsbyInitSchematic",
      "schema": "./src/generators/init/schema.json",
      "description": "Initialize the @nrwl/gatsby plugin",
      "hidden": true
    },
    "application": {
      "factory": "./src/generators/application/application#applicationSchematic",
      "schema": "./src/generators/application/schema.json",
      "aliases": ["app"],
      "x-type": "application",
      "description": "Create an application"
    },
    "component": {
      "factory": "./src/generators/component/component#componentSchematic",
      "schema": "./src/generators/component/schema.json",
      "description": "Create a component"
    },
    "page": {
      "factory": "./src/generators/page/page#pageSchematic",
      "schema": "./src/generators/page/schema.json",
      "description": "Create a page"
    }
  },
  "generators": {
    "init": {
      "factory": "./src/generators/init/init",
      "schema": "./src/generators/init/schema.json",
      "description": "Initialize the @nrwl/gatsby plugin",
      "hidden": true
    },
    "application": {
      "factory": "./src/generators/application/application",
      "schema": "./src/generators/application/schema.json",
      "aliases": ["app"],
      "x-type": "application",
      "description": "Create an application"
    },
    "component": {
      "factory": "./src/generators/component/component",
      "schema": "./src/generators/component/schema.json",
      "description": "Create a component"
    },
    "page": {
      "factory": "./src/generators/page/page",
      "schema": "./src/generators/page/schema.json",
      "description": "Create a page"
    }
  }
}
