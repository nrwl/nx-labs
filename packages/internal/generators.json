{
  "generators": {
    "migrate-to-nx-scope": {
      "factory": "./src/generators/migrate-to-nx-scope/generator",
      "schema": "./src/generators/migrate-to-nx-scope/schema.json",
      "description": "migrate-to-nx-scope generator"
    }
  }
}
