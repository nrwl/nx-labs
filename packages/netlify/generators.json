{
  "$schema": "http://json-schema.org/schema",
  "name": "netlify",
  "version": "0.0.1",
  "generators": {
    "serverless": {
      "factory": "./src/generators/serverless/serverless",
      "schema": "./src/generators/serverless/schema.json",
      "description": "Creates a new project with Netlify serverless configuration",
      "hidden": true
    },
    "setup-functions": {
      "factory": "./src/generators/setup-functions/setup-functions",
      "schema": "./src/generators/setup-functions/schema.json",
      "description": "Creates edge functions that can be deployed to Netlify to an existing project."
    },
    "preset": {
      "factory": "./src/generators/preset/preset",
      "schema": "./src/generators/preset/schema.json",
      "description": "Nx Netlify Serverless preset generator",
      "hidden": true
    }
  }
}
