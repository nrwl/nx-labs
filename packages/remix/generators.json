{
  "$schema": "http://json-schema.org/schema",
  "name": "NxRemix",
  "version": "0.0.1",
  "extends": ["@nrwl/react"],
  "generators": {
    "preset": {
      "implementation": "./src/generators/preset/preset.impl",
      "schema": "./src/generators/preset/schema.json",
      "description": "Generate a new Remix workspace",
      "hidden": true
    },
    "setup": {
      "implementation": "./src/generators/setup/setup.impl",
      "schema": "./src/generators/setup/schema.json",
      "description": "Setup a Remix in an existing workspace",
      "hidden": true
    },
    "application": {
      "implementation": "./src/generators/application/application.impl",
      "schema": "./src/generators/application/schema.json",
      "description": "Generate a new Remix application",
      "aliases": ["app"],
      "x-type": "application"
    },
    "library": {
      "implementation": "./src/generators/library/library.impl",
      "schema": "./src/generators/library/schema.json",
      "description": "Generate a new library",
      "aliases": ["lib"],
      "x-type": "library"
    },
    "route": {
      "implementation": "./src/generators/route/route.impl",
      "schema": "./src/generators/route/schema.json",
      "description": "Generate a new route"
    },
    "resource-route": {
      "implementation": "./src/generators/resource-route/resource-route.impl",
      "schema": "./src/generators/resource-route/schema.json",
      "description": "Generate a new resource route"
    },
    "action": {
      "implementation": "./src/generators/action/action.impl",
      "schema": "./src/generators/action/schema.json",
      "description": "Add an action function to an existing route"
    },
    "loader": {
      "implementation": "./src/generators/loader/loader.impl",
      "schema": "./src/generators/loader/schema.json",
      "description": "Add a loader function to an existing route"
    },
    "style": {
      "implementation": "./src/generators/style/style.impl",
      "schema": "./src/generators/style/schema.json",
      "description": "Generates a new stylesheet and adds it to an existing route"
    },
    "meta": {
      "implementation": "./src/generators/meta/meta.impl",
      "schema": "./src/generators/meta/schema.json",
      "description": "Add a meta function to an existing route"
    },
    "cypress": {
      "implementation": "./src/generators/cypress/cypress.impl",
      "schema": "./src/generators/cypress/schema.json",
      "description": "Generate a project for testing Remix apps using Cypress"
    }
  }
}
