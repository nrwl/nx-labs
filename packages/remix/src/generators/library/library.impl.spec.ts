import { readJson, readProjectConfiguration } from '@nx/devkit';
import { type ProjectNameAndRootFormat } from '@nx/devkit/src/generators/project-name-and-root-utils';
import { createTreeWithEmptyWorkspace } from '@nx/devkit/testing';
import applicationGenerator from '../application/application.impl';
import libraryGenerator from './library.impl';

describe('Remix Library Generator', () => {
  describe.each([
    ['derived', 'libs/test'],
    ['as-provided', 'test'],
  ])(
    '-projectNameAndRootFormat=%s',
    (projectNameAndRootFormat: ProjectNameAndRootFormat, libDir) => {
      it('should generate a library correctly', async () => {
        // ARRANGE
        const tree = createTreeWithEmptyWorkspace({ layout: 'apps-libs' });

        // ACT
        await libraryGenerator(tree, {
          name: 'test',
          style: 'css',
          projectNameAndRootFormat,
        });

        // ASSERT
        const tsconfig = readJson(tree, 'tsconfig.base.json');
        expect(tree.exists(`${libDir}/src/server.ts`));
        expect(tree.children(`${libDir}/src/lib`)).toMatchSnapshot();
        expect(tsconfig.compilerOptions.paths).toMatchSnapshot();
      }, 25_000);

      describe('Standalone Project Repo', () => {
        it('should update the tsconfig paths correctly', async () => {
          // ARRANGE
          const tree = createTreeWithEmptyWorkspace();
          await applicationGenerator(tree, {
            name: 'demo',
            rootProject: true,
          });
          const originalBaseTsConfig = readJson(tree, 'tsconfig.json');

          // ACT
          await libraryGenerator(tree, {
            name: 'test',
            style: 'css',
            projectNameAndRootFormat,
          });

          // ASSERT
          const updatedBaseTsConfig = readJson(tree, 'tsconfig.base.json');
          expect(
            Object.keys(originalBaseTsConfig.compilerOptions.paths)
          ).toContain('~/*');
          expect(
            Object.keys(updatedBaseTsConfig.compilerOptions.paths)
          ).toContain('~/*');
        });
      });

      describe('--unitTestRunner', () => {
        it('should not create config files when unitTestRunner=none', async () => {
          // ARRANGE
          const tree = createTreeWithEmptyWorkspace({ layout: 'apps-libs' });

          // ACT
          await libraryGenerator(tree, {
            name: 'test',
            style: 'css',
            unitTestRunner: 'none',
            projectNameAndRootFormat,
          });

          // ASSERT
          expect(tree.exists(`${libDir}/jest.config.ts`)).toBeFalsy();
          expect(tree.exists(`${libDir}/vite.config.ts`)).toBeFalsy();
        });

        it('should create the correct config files for testing with jest', async () => {
          // ARRANGE
          const tree = createTreeWithEmptyWorkspace({ layout: 'apps-libs' });

          // ACT
          await libraryGenerator(tree, {
            name: 'test',
            style: 'css',
            unitTestRunner: 'jest',
            projectNameAndRootFormat,
          });

          // ASSERT
          expect(
            tree.read(`${libDir}/jest.config.ts`, 'utf-8')
          ).toMatchSnapshot();
          expect(tree.read(`${libDir}/src/test-setup.ts`, 'utf-8'))
            .toMatchInlineSnapshot(`
                    "import { installGlobals } from '@remix-run/node';
                    import '@testing-library/jest-dom/extend-expect';
                    installGlobals();
                    "
                `);
        });

        it('should create the correct config files for testing with vitest', async () => {
          // ARRANGE
          const tree = createTreeWithEmptyWorkspace({ layout: 'apps-libs' });

          // ACT
          await libraryGenerator(tree, {
            name: 'test',
            style: 'css',
            unitTestRunner: 'vitest',
            projectNameAndRootFormat,
          });

          // ASSERT
          expect(
            tree.read(`${libDir}/vite.config.ts`, 'utf-8')
          ).toMatchSnapshot();

          expect(tree.read(`${libDir}/src/test-setup.ts`, 'utf-8'))
            .toMatchInlineSnapshot(`
                    "import { installGlobals } from '@remix-run/node';
                    import '@testing-library/jest-dom/extend-expect';
                    installGlobals();
                    "
                `);
        }, 25_000);
      });

      // TODO(Colum): Unskip this when buildable is investigated correctly
      xit('should generate the config files correctly when the library is buildable', async () => {
        // ARRANGE
        const tree = createTreeWithEmptyWorkspace({ layout: 'apps-libs' });

        // ACT
        await libraryGenerator(tree, {
          name: 'test',
          style: 'css',
          buildable: true,
          projectNameAndRootFormat,
        });

        // ASSERT
        const project = readProjectConfiguration(tree, 'test');
        const pkgJson = readJson(tree, `${libDir}/package.json`);
        expect(project.targets.build.options.format).toEqual(['cjs']);
        expect(project.targets.build.options.outputPath).toEqual(
          `${libDir}/dist`
        );
        expect(pkgJson.main).toEqual('./dist/index.cjs.js');
        expect(pkgJson.typings).toEqual('./dist/index.d.ts');
      });
    }
  );
});
