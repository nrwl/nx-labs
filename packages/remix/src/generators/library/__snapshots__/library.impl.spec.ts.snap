// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Remix Library Generator -projectNameAndRootFormat=as-provided --unitTestRunner should create the correct config files for testing with jest 1`] = `
"/* eslint-disable */
export default {
  setupFilesAfterEnv: ['./src/test-setup.ts'],
  displayName: 'test',
  preset: '../jest.preset.js',
  transform: {
    '^(?!.*\\\\\\\\.(js|jsx|ts|tsx|css|json)$)': '@nx/react/plugins/jest',
    '^.+\\\\\\\\.[tj]sx?$': ['babel-jest', { presets: ['@nx/react/babel'] }],
  },
  moduleFileExtensions: ['ts', 'tsx', 'js', 'jsx'],
  coverageDirectory: '../coverage/test',
};
"
`;

exports[`Remix Library Generator -projectNameAndRootFormat=as-provided --unitTestRunner should create the correct config files for testing with vitest 1`] = `
"import { nxViteTsPaths } from '@nx/vite/plugins/nx-tsconfig-paths.plugin';
import react from '@vitejs/plugin-react';
import { defineConfig } from 'vite';

export default defineConfig({
  cacheDir: '../node_modules/.vite/test',

  plugins: [react(), nxViteTsPaths()],

  // Uncomment this if you are using workers.
  // worker: {
  //  plugins: [ nxViteTsPaths() ],
  // },

  test: {
    setupFiles: ['./src/test-setup.ts'],
    globals: true,
    cache: { dir: '../node_modules/.vitest' },
    environment: 'jsdom',
    include: ['src/**/*.{test,spec}.{js,mjs,cjs,ts,mts,cts,jsx,tsx}'],
  },
});
"
`;

exports[`Remix Library Generator -projectNameAndRootFormat=as-provided should generate a library correctly 1`] = `
Array [
  "test.module.css",
  "test.spec.tsx",
  "test.tsx",
]
`;

exports[`Remix Library Generator -projectNameAndRootFormat=as-provided should generate a library correctly 2`] = `
Object {
  "@proj/test": Array [
    "test/src/index.ts",
  ],
  "@proj/test/server": Array [
    "test/src/server.ts",
  ],
}
`;

exports[`Remix Library Generator -projectNameAndRootFormat=derived --unitTestRunner should create the correct config files for testing with jest 1`] = `
"/* eslint-disable */
export default {
  setupFilesAfterEnv: ['./src/test-setup.ts'],
  displayName: 'test',
  preset: '../../jest.preset.js',
  transform: {
    '^(?!.*\\\\\\\\.(js|jsx|ts|tsx|css|json)$)': '@nx/react/plugins/jest',
    '^.+\\\\\\\\.[tj]sx?$': ['babel-jest', { presets: ['@nx/react/babel'] }],
  },
  moduleFileExtensions: ['ts', 'tsx', 'js', 'jsx'],
  coverageDirectory: '../../coverage/libs/test',
};
"
`;

exports[`Remix Library Generator -projectNameAndRootFormat=derived --unitTestRunner should create the correct config files for testing with vitest 1`] = `
"import { nxViteTsPaths } from '@nx/vite/plugins/nx-tsconfig-paths.plugin';
import react from '@vitejs/plugin-react';
import { defineConfig } from 'vite';

export default defineConfig({
  cacheDir: '../../node_modules/.vite/test',

  plugins: [react(), nxViteTsPaths()],

  // Uncomment this if you are using workers.
  // worker: {
  //  plugins: [ nxViteTsPaths() ],
  // },

  test: {
    setupFiles: ['./src/test-setup.ts'],
    globals: true,
    cache: { dir: '../../node_modules/.vitest' },
    environment: 'jsdom',
    include: ['src/**/*.{test,spec}.{js,mjs,cjs,ts,mts,cts,jsx,tsx}'],
  },
});
"
`;

exports[`Remix Library Generator -projectNameAndRootFormat=derived should generate a library correctly 1`] = `
Array [
  "test.module.css",
  "test.spec.tsx",
  "test.tsx",
]
`;

exports[`Remix Library Generator -projectNameAndRootFormat=derived should generate a library correctly 2`] = `
Object {
  "@proj/libs/test": Array [
    "libs/test/src/index.ts",
  ],
  "@proj/libs/test/server": Array [
    "libs/test/src/server.ts",
  ],
}
`;
